omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -0,0 +1,200 @@
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_2;
+
+import static org.openmrs.ConditionVerificationStatus.CONFIRMED;
+
+import java.util.LinkedHashMap;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Condition;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.Patient;
+import org.openmrs.api.DiagnosisService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.RestTestConstants2_2;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Tests functionality of {@link DiagnosisController2_2}.
+ */
+public class DiagnosisController2_2Test extends MainResourceControllerTest {
+	
+	private DiagnosisService diagnosisService;
+	
+	private Encounter encounter;
+	
+	private Patient patient;
+	
+	private Condition condition;
+	
+	@Before
+	public void before() throws Exception {
+		executeDataSet(RestTestConstants2_2.DIAGNOSIS_TEST_DATA_XML);
+		this.diagnosisService = Context.getDiagnosisService();
+		this.encounter = Context.getEncounterService().getEncounter(1);
+		this.patient = Context.getPatientService().getPatient(1);
+		this.condition = Context.getConditionService().getCondition(1);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getURI()
+	 */
+	@Override
+	public String getURI() {
+		return "patientdiagnoses";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_2.DIAGNOSIS_UUID;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return diagnosisService.getPrimaryDiagnoses(this.encounter).size();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#shouldGetAll()
+	 */
+	@Override
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldGetAll() throws Exception {
+		super.shouldGetAll();
+	}
+	
+	@Test
+	public void shouldCreateADiagnosis() throws Exception {
```

Comment:
Can you rename this to shouldCreateANonCodedDiagnosis?
Then create a second test called shouldCreateACodedDiagnosis